pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go

```diff
@@ -31,32 +33,204 @@ type ResourceInterpreterCustomizationSpec struct {
 	// +required
 	Kind string `json:"kind"`
```

Comment:
I don't think put `APIVersion` and `Kind` in this struct is a good thing, for the reasons:
1. `APIVersion` and `Kind` are not customization rulers
2. The spec will be very large, and users will be difficult to find the these two field. These fields are important for users.

I prefer the format:
```yaml
apiVersion: config.karmada.io/v1alpah1
kind: ResourceInterpreterCustomization
metadata:
  name: foo
target:
  apiVersion: v1
  kind: Pod
customizations:
  ...
```

Reply:
Good idea.

Reply:
Good point. But I'm afraid the `target` might be present always after `customizations`(`Alphabetical order`).
Is that OK? Do we have other alternatives for `target`?
```yaml
customizations:
target:
  apiVersion: v1
  kind: Pod
```

Reply:
accept.